### PR TITLE
Update metadata-stream.js so TXXX are not clipped

### PR DESCRIPTION
**Pull request by RevinKey**
## Description
Parsing of TXXX tag is cutting off last character of tag. 
Do not need to account for a null terminator.

## Sample with Brightcove (videojs)player:
current: http://cs1.brightcodes.net/kreynolds/BrightcoveExamples/test/414473/current.html
proposed: http://cs1.brightcodes.net/kreynolds/BrightcoveExamples/test/414473/index.html

## Specific Changes proposed
change line48 in file metadata-stream.js to 
          tag.value = parseUtf8(tag.data, i + 1, tag.data.length).replace('\0','');

## Requirements Checklist
- [ ] Feature implemented 
- [ ] If necessary, more likely in a feature request than a bug fix
  - [x] Example created ([starter template on JSBin](http://jsbin.com/axedog/edit?html,output))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by imbcmdth** [comment]
The null-terminator **may** be present according to the spec. There are instances where it should be ignored (such as in the test cases that are failing). The code should check for and remove any null-termination before processing the string.

**Comment by RevinKey** [comment]
Would this be enough for parsing out the null character:
 tag.value = parseUtf8(tag.data, i + 1, tag.data.length).replace('\0','');

In my test it works. I updated my player script in my sample to reflect this change as well if you would like to view it in action.

**Comment by imbcmdth** [comment]
Thanks!

Looks like it works but I worry about multiple trailing nulls. That replace will only replace the first instance of `\0` it finds in the string. I think a better solution might be `.replace(/\0*$/, '');` to replace *one or more* null characters that appear at the **end** of the string.

**Comment by RevinKey** [comment]
Cool. My Branch is updated and passed Travis. Let me know if you need anything else.

File: lib/m2ts/metadata-stream.js [hunk 1]
```diff
@@ -5,243 +5,246 @@
...
+                return;
+            }
+
+            for (i = 1; i < tag.data.length; i++) {
+                if (tag.data[i] === 0) {
+                    // parse the text fields
+                    tag.description = parseUtf8(tag.data, 1, i);
+
+
+                    //do not include the null terminator in the tag value
```
Comment (imbcmdth) [inline lib/m2ts/metadata-stream.js]: This is failing linting in travis because it's missing a space after the `//`.

**Comment by imbcmdth** [comment]
@RevinKey Seems like the indenting on the file changed and there are a few extra newlines (surprised that the linting didn't complain about either!)

Compare: https://github.com/videojs/mux.js/pull/148/files
With: https://github.com/videojs/mux.js/pull/148/files?w=1

And you can see how many lines had nothing changed except newlines. Would you mind getting this PR back down to 2 or three lines changed?

File: lib/m2ts/metadata-stream.js [hunk 2]
```diff
@@ -5,243 +5,246 @@
...
+                // ignore frames with unrecognized character encodings
+                return;
+            }
+
+            for (i = 1; i < tag.data.length; i++) {
+                if (tag.data[i] === 0) {
+                    // parse the text fields
+                    tag.description = parseUtf8(tag.data, 1, i);
+
+
```
Comment (imbcmdth) [inline lib/m2ts/metadata-stream.js]: Extra newline.

**Comment by imbcmdth** [comment]
@RevinKey Could you add a test for the non-null terminating case? It should be pretty simple to take an existing test and creating a new one from that. As for all the whitespace changes, it might be easier to just start over and do the change again.
